File: exercise/java/day02/src/main/java/games/FizzBuzz.java
package games;

public class FizzBuzz {

    public static final int BUZZ_MULTIPLIER = 5;
    public static final int FIZZ_MULTIPLIER = 3;

    private FizzBuzz() {
    }
    
    public static String convert(Integer input) throws OutOfRangeException {
        if (isOutOfRange(input)) {
            throw new OutOfRangeException();
        }

        return convertIntegerToFizzBuzz(input);
    }

    private static String convertIntegerToFizzBuzz(Integer input) {
        if (isMultipleOf(input, BUZZ_MULTIPLIER) && isMultipleOf(input, FIZZ_MULTIPLIER)) {
            return "FizzBuzz";
        }
        if (isMultipleOf(input, FIZZ_MULTIPLIER)) {
            return "Fizz";
        }
        if (isMultipleOf(input, BUZZ_MULTIPLIER)) {
            return "Buzz";
        }

        return input.toString();
    }

    private static boolean isOutOfRange(Integer input) {
        return input <= 0 || input >= 100;
    }

    private static boolean isMultipleOf(Integer input, Integer modulo) {
        return input % modulo == 0;
    }
}

File: exercise/kotlin/day06/src/main/kotlin/games/FizzBuzz.kt
package games

private const val MIN = 0
private const val MAX = 100
private const val FIZZBUZZ = 15
private const val FIZZ = 3
private const val BUZZ = 5

object FizzBuzz {
    fun convert(input: Int): String {
        if (isOutOfRange(input)) throw OutOfRangeException()
        return convertSafely(input)
    }

    private fun convertSafely(input: Int): String {
        return when {
            `is`(FIZZBUZZ, input) -> "FizzBuzz"
            `is`(FIZZ, input) -> "Fizz"
            `is`(BUZZ, input) -> "Buzz"
            else -> input.toString()
        }
    }

    private fun `is`(divisor: Int, input: Int): Boolean {
        return input % divisor == 0
    }

    private fun isOutOfRange(input: Int) = input <= MIN || input > MAX
}